hail/hail/src/is/hail/services/BatchClient.scala

```diff
@@ -309,8 +309,7 @@ case class BatchClient private (req: Requester) extends Logging with AutoCloseab
     while (true) {
       val jobGroup = getJobGroup(batchId, jobGroupId)
 
-      if (jobGroup.complete)
-        return jobGroup
+      if (jobGroup.state != JobGroupStates.Running) return jobGroup
```

Comment:
Are these enumerated anywhere? Are we sure there's not a `Ready`?

Reply:
```
  case object Failure extends JobGroupState
  case object Cancelled extends JobGroupState
  case object Success extends JobGroupState
  case object Running extends JobGroupState
```

... yes

Reply:
These are reimplemented from batch/front_end.py. fwiw I don't like this solution and feels fragile.